scripts/zones/Lower_Jeuno/npcs/Zalsuhm.lua

```diff
@@ -3,114 +3,109 @@
+            if player:getVar("Upset_Zalsuhm") > 0 then
+                player:setVar("Upset_Zalsuhm", 0)
             end
 
-            if (nyzulWeapon) then -- The player has a Nyzul weapon in the mainHand, try to initiate quest
-                player:startEvent(10086, mainJobId);
+            if nyzulWeapon then
+                player:startEvent(10086, mainJobId)
             else
-                player:startEvent(10085); -- Default dialog
+                player:startEvent(10085)
             end
         end
-    elseif (unlockingAMyth == QUEST_ACCEPTED) then -- Quest is active for current job
-        player:startEvent(10087); -- Zalsuhm asks for the player to show him the weapon if they sense a change
-    else -- Quest is complete for the current job
-        player:startEvent(10089);
+    elseif unlockingAMyth == QUEST_ACCEPTED then
+        player:startEvent(10087)
+    else
+        player:startEvent(10089)
     end
-    
-end;
+end
 
 function onEventUpdate(player,csid,option)
-
-    -- printf("LowerJeuno_Zalsuhm.onEventUpdate() - ");
-    -- printf("\tCSID: %u", csid);
-    -- printf("\tRESULT: %u", option);
-
-end;
+end
 
 function onEventFinish(player,csid,option)
-
-    -- printf("LowerJeuno_Zalsuhm.onEventFinish() - ");
-    -- printf("\tCSID: %u", csid);
-    -- printf("\tRESULT: %u", option);
-
-    -- Zalsuhm wants to research the player's Nyzul Weapon
-    if (csid == 10086) then
-        -- The player chose "He has shifty eyes" (turns down the quest)
-        if (option == 53) then
-            player:setVar("Upset_Zalsuhm", 1);
-            player:needToZone(true);
-        elseif (option <= dsp.job.SCH) then -- Just to make sure we didn't get into an invalid state
-            -- The player chose "More power" (accepts the quest)
-            local questId = getQuestId(option);
-            player:addQuest(JEUNO, questId);
+    local questId = getQuestId(option)
+    if csid == 10086 then
+        if option == 53 then
+            player:setVar("Upset_Zalsuhm", 1)
+            player:needToZone(true)
+        elseif option <= dsp.job.SCH then
+            player:addQuest(JEUNO, questId)
         end
-    elseif (csid == 10088 and option <= dsp.job.SCH) then -- The quest is completed
-        local questId = getQuestId(option);
-        player:completeQuest(JEUNO, questId);
+    elseif csid == 10088 and option <= dsp.job.SCH then
+        local jobs = 
+        {
+            [dsp.job.WAR] = 24,
+            [dsp.job.MNK] = 15,
+            [dsp.job.WHM] = 29,
+            [dsp.job.BLM] = 30,
+            [dsp.job.RDM] = 20,
+            [dsp.job.THF] = 17,
+            [dsp.job.PLD] = 21,
+            [dsp.job.DRK] = 25,
+            [dsp.job.BST] = 23,
+            [dsp.job.BRD] = 18,
+            [dsp.job.RNG] = 33,
+            [dsp.job.SAM] = 28,
+            [dsp.job.NIN] = 27,
+            [dsp.job.DRG] = 26,
+            [dsp.job.SMN] = 31,
+            [dsp.job.BLU] = 22,
+            [dsp.job.COR] = 34,
```

Comment:
34 > 32

Reply:
You're confusing the value with the key. The 34 is the weaponskill id that COR (job id 17) can unlock.

Reply:
Those are the unlock keys i got from the weaponskill.sql to unlock said weaponskill for said job

Reply:
oups